### DISCUSSION TITLE
What's the procedure to ask (or help) on package upgrade ?

### DISCUSSION THREAD

**Discussion by ScreamZ** [expo/expo#18568] [Archived: Questions] [answered]
hello, I'm experiencing a lot of 
```
Some dependencies are incompatible with the installed expo package version:
 - @stripe/stripe-react-native - expected version: 0.6.1 - actual version installed: 0.9.0
 - expo-dev-client - expected version: ~1.0.1 - actual version installed: 1.0.0
 - expo-notifications - expected version: ~0.15.4 - actual version installed: 0.15.2
 - expo-updates - expected version: ~0.13.4 - actual version installed: 0.13.2
Your project may not work correctly until you install the correct versions of the packages.
To install the correct versions of these packages, please run: expo doctor --fix-dependencies,
or install individual packages by running expo install [package-name ...]
```

There I have two topics:

- How can I get the information or whether upgrading to recommend version will require some native code delivery (and if I can codepush using EAS update or not ?) Sometimes recommended version changes within the same expo SDK version.
- Lastly, how can we help upgrading to a more recent version? Can't see anywhere we can make pull request for that ? https://github.com/stripe/stripe-react-native is at the day v0.16, so I bypassed by hand to match latest features, but even though it's stuck on 0.6.1 on expo recommedations
- 
Thanks

**Comment by raajnadar** [comment]
You can use any version with the expo dev client & to fix the warnings run `expo doctor --fix-dependencies` it will install compatible packages as per your current SDK version

**Reply by ScreamZ** [reply]
Yes but this is not the most up to date package, only one you added as compatible. Later version can be compatible, that's why I asked for the procedure to upgrade this "suggestion"

**Reply by raajnadar** [reply]
Expo SDK 46 uses the 0.13.1 version of stripe SDK

https://github.com/expo/expo/pull/18072

You can see the How section of this PR to understand how the maintainer upgraded and tested the module

**Comment by keith-kurak** [comment, marked as answer]
Suggested versions are tailored to your SDK version, and it should not suggest that you upgrade to a version that is incompatible with your SDK. Within an SDK version, it may suggest upgrading to a version that is compatible with native the previous version's native code. For instance, recently with SDK 46, `expo doctor --fix-dependencies` would update `react-native` from 0.69.3 to 0.69.4, since there are no native code changes, only JavaScript.

These version numbers generally err on the side of caution to ensure support with the Expo Go app. So, if you are using a EAS Build/ development client, then it's quite possible that you can safely use a newer version. At that point, the potential blockers are things like the version of React Native, Android API level, etc. And, of course, if you know that a more recent version only changed JS code, then that's pretty safe, even with Expo Go.

If you see a suggested version that you believe is in error, open up an issue to let us know. We update this information in a few different places to ensure it's accurate for the specific SDK version.

**Reply by ScreamZ** [reply]
Hello Keith, 
Thanks for this specific answer :-)

Have a nice day

**Reply by DevSrabon** [reply]
expo doctor --fix-dependencies doesn't support in sdk49.
I tried npx expo-doctor@latest. But it doesn't fix
 [stderr] 

Note: /home/expo/workingdir/build/node_modules/@react-native-async-storage/async-storage/android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageModule.java uses or overrides a deprecated API.

**Comment by DevSrabon** [comment]
expo doctor --fix-dependencies doesn't support in sdk49.
I tried npx expo-doctor@latest. But it doesn't fix
[stderr]

Note: /home/expo/workingdir/build/node_modules/@react-native-async-storage/async-storage/android/src/main/java/com/reactnativecommunity/asyncstorage/AsyncStorageModule.java uses or overrides a deprecated API.
